src/channel.ts

```diff
@@ -1496,9 +1497,14 @@ export class Channel<
           s.watchers = s.watchers.without(event.user.id);
         }
         break;
+      case 'message.deleted':
+        if (event.message) {
+          if (event.hard_delete) s.removeMessage(event.message as { id: string });
```

Comment:
casting because id in MessageBase is possibly undefined?

Reply:
yea, it's either checking the id for undefined(which somehow TS was failing to understand) or casting the message type

Reply:
i guess you could do `if (event.hard_delete && event.message.id)` if we don't want to change the type

Reply:
weirdly yea